File: PortalApp.API/Models/Navig.cs
using System;
using System.Collections.Generic;

namespace PortalApp.API.Models
{
    public class Navig
    {
        public Guid? Id { get; set; }
        public string Title { get; set; }
        public string Type { get; set; }
        public string Icon { get; set; }
        public string Url { get; set; }
        public string TitleEng { get; set; }
        public string TitleKaz { get; set; }
        
        public ICollection<Navig> Children { get; set; }
    }
}

File: PortalApp-SPA/src/app/_models/navig.ts
export class Navig {
    id: string;
    title: string;
    type: string;
    icon: string;
    url: string;
    children?: Navig[];
    expanded: boolean;
    selected: boolean;

    constructor(navig?) {
        navig = navig || {};
        this.id = navig.id || '';
        this.title = navig.title || '';
        this.type = navig.type || '';
        this.icon = navig.icon || '';
        this.url = navig.url || '';
        this.children = navig.children || [];
        this.expanded = navig.expanded || false;
        this.selected = navig.selected || false;
    }
}
